src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookSearch.ts

```diff
@@ -0,0 +1,686 @@
+import { MessageType } from 'vs/base/browser/ui/inputbox/inputBox';
+import * as aria from 'vs/base/browser/ui/aria/aria';
+import * as errors from 'vs/base/common/errors';
+import { NotebookSearchWidget } from 'sql/workbench/contrib/notebook/browser/notebookExplorer/notebookSearchWidget';
+import { ITreeElement, ITreeContextMenuEvent } from 'vs/base/browser/ui/tree/tree';
+import { Iterable } from 'vs/base/common/iterator';
+import { searchClearIcon, searchCollapseAllIcon, searchExpandAllIcon, searchStopIcon } from 'vs/workbench/contrib/search/browser/searchIcons';
+import { Action, IAction } from 'vs/base/common/actions';
+import { createAndFillInContextMenuActions } from 'vs/platform/actions/browser/menuEntryActionViewItem';
+import { Memento } from 'vs/workbench/common/memento';
+
+const $ = dom.$;
+
+export class NotebookSearchView extends SearchView {
+	static readonly ID = 'notebookExplorer.searchResults';
+
+	private treeSelectionChangeListener: IDisposable;
+
+	private viewActions: Array<CollapseDeepestExpandedLevelAction | ClearSearchResultsAction> = [];
+	private cancelSearchAction: CancelSearchAction;
+	private toggleExpandAction: ToggleCollapseAndExpandAction;
+
+	constructor(
+		options: IViewPaneOptions,
+		@IFileService readonly fileService: IFileService,
+		@IEditorService readonly editorService: IEditorService,
+		@IProgressService readonly progressService: IProgressService,
+		@INotificationService readonly notificationService: INotificationService,
+		@IDialogService readonly dialogService: IDialogService,
+		@IContextViewService readonly contextViewService: IContextViewService,
+		@IInstantiationService instantiationService: IInstantiationService,
+		@IViewDescriptorService viewDescriptorService: IViewDescriptorService,
+		@IConfigurationService configurationService: IConfigurationService,
+		@IWorkspaceContextService readonly contextService: IWorkspaceContextService,
+		@ISearchWorkbenchService readonly searchWorkbenchService: ISearchWorkbenchService,
+		@IContextKeyService readonly contextKeyService: IContextKeyService,
+		@IReplaceService readonly replaceService: IReplaceService,
+		@ITextFileService readonly textFileService: ITextFileService,
+		@IPreferencesService readonly preferencesService: IPreferencesService,
+		@IThemeService themeService: IThemeService,
+		@ISearchHistoryService readonly searchHistoryService: ISearchHistoryService,
+		@IContextMenuService contextMenuService: IContextMenuService,
+		@IMenuService readonly menuService: IMenuService,
+		@IAccessibilityService readonly accessibilityService: IAccessibilityService,
+		@IKeybindingService keybindingService: IKeybindingService,
+		@IStorageService storageService: IStorageService,
+		@IOpenerService openerService: IOpenerService,
+		@ITelemetryService telemetryService: ITelemetryService,
+		@ICommandService readonly commandService: ICommandService,
```

Comment:
nit: comma

Reply:
I'll clean this up in a separate PR with my extensible changes.